docs/csharp/language-reference/tokens/interpolated.md

```diff
@@ -86,6 +86,23 @@ If an interpolated string has the type `string`, it's typically transformed into
 
 If an interpolated string has the type <xref:System.IFormattable> or <xref:System.FormattableString>, the compiler generates a call to the <xref:System.Runtime.CompilerServices.FormattableStringFactory.Create%2A?displayProperty=nameWithType> method.
 
+> [!NOTE]
+> String interpolation can cause [boxing](../../programming-guide/types/boxing-and-unboxing.md), as it happens in the following example:
+>
+> ```csharp-interactive
+> double pi = Math.PI;
+> string line = $"The pi constant is {pi:F5}";
+> Console.WriteLine(line);  // output: The pi constant is 3.14159
+> ```
+>
+> If you need to avoid boxing, format an instance of a value type explicitly, as the following example shows:
```

Comment:
Avoiding boxing is not a goal by itself, avoiding heap allocations is. And as I showed in https://github.com/dotnet/docs/issues/16618#issuecomment-573315069, using `ToString()` this way can result in *more* allocations. So I think I would add some hand-waving and say something like:

> If you need to avoid heap allocations caused by boxing, it can be advantageous to format an instance of a value type explicitly, as the following example shows:

Also, it could make sense to use a type where `ToString()` actually avoids allocations on a recent version of .Net Core. I believe any `enum` type would be suitable for this purpose:

* it's a value type,
* it doesn't implement `ISpanFormattable` (which is the interface that helps avoiding intermediate string allocations),
* it supports format strings.

Reply:
@svick thanks; I've updated the note to provide more context